inc/block-editor.php

```diff
@@ -0,0 +1,63 @@
+<?php
+/**
+ * Block editor (gutenberg) specific functionality
+ *
+ * @package UnderStrap
+ */
+
+
+add_action( 'after_setup_theme', 'understrap_block_editor_setup' );
+
+if ( ! function_exists( 'understrap_block_editor_setup' ) ) {
+
+	/**
+	 * Sets up our default theme support for the WordPress block editor.
+	 *
+	 */
+	function understrap_block_editor_setup() {
+
+		// Add support for the block editor stylesheet.
+		add_theme_support( 'editor-styles' );
+
+		// Add support for wide alignment.
+		add_theme_support( 'align-wide' );
+
+		// Register our custom colors as options in the editor.
+		$color_palette = understrap_generate_color_palette();
+		if ( $color_palette ) {
+			add_theme_support( 'editor-color-palette', $color_palette );
+		}
+	}
+}
+
+if ( ! function_exists( 'understrap_generate_color_palette' ) ) {
+	/**
+	 * Checks for our JSON file of color values. If exists, creates a color palette array.
+	 *
+	 * @return array|bool
```

Comment:
When does the function have a boolean return type?

Reply:
If they've chosen to override with a filter it could theoretically return false to cancel it out - although I guess if they want to override it they should return an empty array....

Reply:
You're not type casting the return value. It can be whatever the user wants it to be ;-)